lib/rucio/core/rule.py

```diff
@@ -143,10 +143,10 @@ def default(rule: models.ReplicationRule, did: models.DataIdentifier, session: '
 
 
 @transactional_session
-def add_rule(dids, account, copies, rse_expression, grouping, weight, lifetime, locked, subscription_id,
-             source_replica_expression=None, activity='User Subscriptions', notify=None, purge_replicas=False,
-             ignore_availability=False, comment=None, ask_approval=False, asynchronous=False, ignore_account_limit=False,
-             priority=3, delay_injection=None, split_container=False, meta=None, *, session: "Session", logger=logging.log):
+def add_rule(dids: Sequence[dict[str, Any]], account: InternalAccount, copies: int, rse_expression: str, weight: str, lifetime: int, locked: bool, subscription_id: str,
```

Comment:
Note: position change of `grouping` due to it becoming non-default argument. I checked all `add_rule` calls to make sure that every time the arguments are named every time the function is called.

Reply:
I've reverted this change in latest commit as it caused some issues - these were likely not due to the argument becoming non-default, but rather about `grouping` being passed as a `RuleGrouping` rather than a `str`.